homeassistant/components/ozw/light.py

```diff
@@ -133,3 +217,47 @@ async def async_turn_off(self, **kwargs):
         self.async_set_duration(**kwargs)
 
         self.values.primary.send_value(0)
+
+    def _calculate_rgb_values(self):
+        # Color Channels
+        self._color_channels = self.values.color_channels.data[ATTR_VALUE]
+
+        # Color Data String
+        data = self.values.color.data[ATTR_VALUE]
+
+        # RGB is always present in the openzwave color data string.
+        rgb = [int(data[1:3], 16), int(data[3:5], 16), int(data[5:7], 16)]
+        self._hs = color_util.color_RGB_to_hs(*rgb)
+
+        # Parse remaining color channels. Openzwave appends white channels
+        # that are present.
+        index = 7
+        temp_1 = 0
+        temp_2 = 0
+
+        # Warm white
+        if self._color_channels & COLOR_CHANNEL_WARM_WHITE:
+            self._white = int(data[index : index + 2], 16)
+            temp_1 = self._white
```

Comment:
Maybe rename `temp_1` to `temp_warm` and `temp_2` to `temp_cold` or similar?

Reply:
updated in latest commit